### PR TITLE
Initial documentation

### PR DESCRIPTION
**Pull request by xzyfer**
This pull request starts the process of documenting Asimov. The documentation will initially be a collection of markdown files, but will eventually be an [assemble.io](assemble.io) driven static site.

I've left placeholders for theme documentation, but they will not be covered initially.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,169 @@
...
+
+Asimov is a rethinking of frontend frameworks in a modern style. The aim is to move away from today's all encompassing frameworks to a modular components approach.
+
+# Table of Contents
+
+1. [Why Asimov](#why-asimov)
+  * [The problem](#the-problem)
+  * [Our solution](#our-solution)
+  * [Built for today](#build-for-today)
+    * [Technology behind Asimov](#technology-behing-asimov)
```
Comment (jamosonic) [inline README.md]: "behing" => "behind"

File: README.md [hunk 2]
```diff
@@ -1,4 +1,169 @@
...
+    * [Components](#theme-components)
+    * [Output](#theme-output)
+  * [Writing your first theme](#theme-create)
+7. [FAQ](#faq)
+  * [Why the strange import paths?](#why-the-strange-import-paths)
+8. [License and Acknowledgements](#license-and-acknowledgements)
+
+# Why Asimov
+
+Over recent years we've experiences rapid increase in the complexity of frontend engineering.
```
Comment (jamosonic) [inline README.md]: "experiences" => "experienced"

File: README.md [hunk 3]
```diff
@@ -1,4 +1,169 @@
...
+
+With the increased complexity grew a need for ways to easily build good looking responsive sites. A lot of projects have taken on the challenge, with a couple noteable successes in [bootstrap](http://getboostrap.com) and [foundation](http://foundation.zurb.com).
+
+Asimov is a re-thinking of current frontend frameworks that attempts to address some of the [new problems](#problems) we've created.
+
+## The problem
+
+Traditionally frontend frameworks have taken the kitchen sink approach. What starts off as a humble set of sane defaults, grows over time into forms, dropdowns, modals etc.. The larger a framework gets the more difficult it becomes to use only certain parts of a framework.
+
+As this growth happens a framework will inevitably develop it's own theme which can be hard to separate from it's individual components. That coupling between theme and structure can, in some cases, make developing your own distinct visual style very difficult. The result is your site looks notably similiar to many others.
```
Comment (jamosonic) [inline README.md]: Lose the apostrophe from "it's". Only use "it's" if you can say "it is" and still have it make sense. Other wise it's just its.

**Comment by xzyfer** [comment]
Clearing out my open pull requests. This is repos is not used.

**Pull request closed without merging**